libparsec/crates/client/tests/unit/client/update_user_profile.rs

```diff
@@ -0,0 +1,224 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use std::sync::Arc;
+
+use libparsec_client_connection::test_register_send_hook;
+use libparsec_protocol::authenticated_cmds;
+use libparsec_tests_fixtures::prelude::*;
+use libparsec_types::prelude::*;
+
+use crate::{Client, ClientUserUpdateProfileError};
+
+use super::utils::client_factory;
+
+async fn update_profile_and_check(profile: UserProfile, user_id: UserID, client: &Arc<Client>) {
+    client.update_user_profile(user_id, profile).await.unwrap();
+
+    let users = client.list_users(false, None, None).await.unwrap();
+    let bob = users.iter().find(|user| user.id == user_id).unwrap();
+    p_assert_eq!(bob.current_profile, profile);
```

Comment:
This helper function is generic so it should be `user` here:
```suggestion
    let user = users.iter().find(|user| user.id == user_id).unwrap();
    p_assert_eq!(user.current_profile, profile);
```